### PR TITLE
Add Support for Loading Environment Variables from .env Files

### PR DESCRIPTION
**Pull request by JamborJan**
Hello `Telmate/proxmox-api-go` maintainers,

I am just starting to contribute to the Proxmox ecosystem. I want to see a nice and useful terraform provider and do my part to it. As I learned now the `Telmate/proxmox-api-go` is the first crucial thing and I cloned the repo to play around with it to get started.

**My Motivation:**
When it came to exporting environment variables to setup the configuration, I thought `.env` files would be a nice thing to have. I usually have multiple small projects ongoing at the same time and always having to type `export KEY=value` is error prone and as it is part of the bash history also insecure.

**Implementation:**
To address this, I have integrated the `godotenv` package, which allows the application to automatically load environment variables from a `.env` file at runtime. This means users can now store their sensitive credentials safely in `.env` files, significantly reducing the risk of accidental exposure, mixing up things etc. I also added `.env` to the gitignore file, everybody should know that this is a sensitive file which must not be shared.

**Changes Made:**

- Added `godotenv` as a dependency in `go.mod`.
- Implemented `.env` file loading logic in the main initialization section of the application.
- Updated the README.md to include instructions on how to use `.env` files with the Proxmox API Go client.

Thank you for considering my PR.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -16,13 +16,21 @@ go build -o proxmox-api-go
 
 ## Run
 
+Create a local `.env` file in the root directory of the project and add the following environment variables:
```
Comment (mleone87) [inline README.md:19]: @JamborJan I would leave the user the choice to keep with the classic `export` way along with .env loading

Comment (JamborJan) [reply README.md:19]: Thanks for pointing that out @mleone87. It was not my intention to remove that functionality but I tested wrong and didn't catch that one. I fixed it now and some manually testing was successful. The projects test are not able yet to include these aspects of testing. Maybe that's something for the future and we should open an issue for that. Not sure if #210 is the right place for that.

Let me know if you have any further feedback. Thanks again for considering my PR.

**Comment by mleone87** [comment]
@JamborJan please fix linting issues

**Comment by JamborJan** [comment]
Hi @mleone87, I guess you asked me to fix the staticcheck error. I did that. Fixing the linting would involve a ton of work as the whole project is not properly setup. I can do that. But I would prefer to do this in a separate issue / pull request to not have scope creep in this specific PR. Are you okay with that? Thank you.

**Comment by mleone87** [comment]
@JamborJan it's ok I was just referring to the lint error on this commit, other are warnings!

**Comment by JamborJan** [comment]
Thank you @mleone87 . Anyway if we want to get rid of the warnings I can do that. Sort of a probation to get my credits for becoming a proper contributor 😅

**Pull request merged**